### PR TITLE
Documentation: Remove mentions of the users@ and devel@ mailing lists.

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

The users@ list doesn't exist anymore (is in autoreply mode to notify users). The devel@ address points to the forum category and should not be advertised as mailing list.

### Testing procedure

Read the changes maybe :sweat_smile: 

### Issues/PRs references

See [here](https://forum.riot-os.org/t/start-a-topic-via-email-which-email-addresses/3082/6)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -191,8 +190,7 @@ into the source repository.
   preoccupied with other PRs. If it happens that your PR receives no review for
   a long time, don't hesitate to gently solicit a review by commenting or
   by explicitly mentioning a maintainer that you know is knowledgeable in the
-  area of the PR. You can also advertise the PR on devel@riot-os.org mailing
-  list and ask for a review there.
+  area of the PR.
```
Comment (miri64) [inline CONTRIBUTING.md:193]: Why not add a link to the forum here as well?

Comment (bergzand) [reply CONTRIBUTING.md]: Sounds good, I've reworded the removed sentence to mention the forum.

**Review by jeandudey — APPROVED** [review]
ACK

Feel free to squash

**Comment by bergzand** [comment]
Squashed!

**Pull request merged**